aten/src/ATen/ScalarTypeUtils.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "ATen/ScalarType.h"
+
+namespace at {
+
+template <typename T>
+struct CTypeToScalarType {
+  static at::ScalarType to();
```

Comment:
I don't think you want this function declaration. It will cause linker errors (instead of errors at compile time) if someone uses `CTypeToScalarType` on an unsupported type.

Reply:
Right, good point.